File: problems/dolls.cpp
#include <bits/stdc++.h>
using namespace std;

bool comp (const pair<int, int>& p1, const pair<int, int>& p2) {
  if (p1.first == p2.first) {
    return p1.second < p2.second;
  }
  return p1.first > p2.first;
}

int main() {
  int T, C, w, h;
  cin >> T;
  while (T--) {
    int C;
    cin >> C;
    vector<pair<int, int> > dolls;
    for (int i = 0; i < C; ++i) {
      cin >> w >> h;
      dolls.emplace_back(w, h);
    }
    sort(dolls.begin(), dolls.end(), comp);

    vector<pair<int, int> > stacks;
    for (auto _doll = dolls.begin(); _doll != dolls.end(); ++_doll) {
      bool b = false;
      for (auto _stk = stacks.begin(); _stk != stacks.end(); ++_stk) {
        if (_stk->first > _doll->first && _stk->second > _doll->second) {
          b = true;
          _stk->first = _doll->first;
          _stk->second = _doll->second;
          break;
        }
      }
      if (!b) {
        stacks.push_back(*_doll);
      }
    }

    cout << stacks.size() << '\n';
  }
}

File: problems/dolls.py
T = int(input())
out = []

for _ in range(T):
    c = int(input())*2
    s = list(map(int, input().split()))
    dolls = []
    for i in range(0,c,2):
        dolls.append([s[i], s[i+1]])
    dolls.sort(key= lambda x: (-x[0], x[1]))
    stacks = []
    for d in dolls:
        b = False
        for s in stacks:
            if s[0] > d[0] and s[1] > d[1]:
                b = True
                s[0], s[1] = d[0], d[1]
                break
        if not b:
            stacks.append(d)

    out.append(str(len(stacks)))


print("\n".join(out))
